infrastructure/aws/cdk/app.py

```diff
@@ -242,10 +242,10 @@ def __init__(  # noqa: C901
                 },
             )
 
-            if eostac_settings.stac_browser_github_tag is not None:
+            if eostac_settings.stac_api_custom_domain_name is not None:
                 assert (
-                    eostac_settings.stac_api_custom_domain_name is not None
-                ), "stac_api_custom_domain_name must be set if stac_browser_github_tag is not None. The browser deployment needs a resolved STAC API url at deployment time and so needs to rely on a predefined custom domain name."
+                    eostac_settings.stac_browser_github_tag is not None
+                ), "stac_browser_github_tag must be set if stac_api_custom_domain_name is not None."
```

Comment:
@emileten can you explain again the relationship between the `custom domain name` and the `GitHub tag`?

Reply:
sorry, the way I wrote it, it's confusing. 

In `eoSTACSettings`, I defined two parameters for the purpose of the STAC browser : 

- the catalog URL, which I named `stac_api_custom_domain_name`, that the browser is going to show
- the version of the browser, `stac_browser_github_tag`

These two things should probably go in another configuration class named `eoStacBrowserSettings` for clarity. 

And the assertion just checks that a version is provided if a catalog url is provided to avoid unclear errors downstream. 

Does that make sense @vincentsarago ?

Reply:
👍 thanks 

> These two things should probably go in another configuration class named eoStacBrowserSettings for clarity.

That would be lovely